### PR TITLE
fix(syncserver): Configure syncserver to use localhost verifier.

### LINKED ISSUE #112: Unable to sync  - Failed to GET https /.well-known/browserid
Currently, when attempting to sync using this repo, the sync server returns

```
Traceback (most recent call last):
  File "/Users/vijaybudhram/Desktop/fxa-local-dev/syncserver/local/lib/python2.7/site-packages/tokenserver/views.py", line 110, in _valid_browserid_assertion
    assertion = verifier.verify(assertion)
  File "/Users/vijaybudhram/Desktop/fxa-local-dev/syncserver/local/lib/python2.7/site-packages/tokenserver/verifiers.py", line 92, in verify
    data = super(LocalBrowserIdVerifier, self).verify(assertion, audience)
  File "/Users/vijaybudhram/Desktop/fxa-local-dev/syncserver/local/lib/python2.7/site-packages/browserid/verifiers/local.py", line 115, in verify
    cert = self.verify_certificate_chain(certificates, now=now)
  File "/Users/vijaybudhram/Desktop/fxa-local-dev/syncserver/local/lib/python2.7/site-packages/browserid/verifiers/local.py", line 161, in verify_certificate_chain
    root_key = self.supportdocs.get_key(root_issuer)
  File "/Users/vijaybudhram/Desktop/fxa-local-dev/syncserver/local/lib/python2.7/site-packages/browserid/supportdoc.py", line 63, in get_key
    supportdoc = self.get_support_document(hostname)
  File "/Users/vijaybudhram/Desktop/fxa-local-dev/syncserver/local/lib/python2.7/site-packages/browserid/supportdoc.py", line 58, in get_support_document
    raise error
ConnectionError: Failed to GET https://127.0.0.1:9000/.well-known/browserid. Reason: EOF occurred in violation of protocol (_ssl.c:661)
```

I have tried updating the sync docker startup script to use the new `SYNCSERVER_IDENTITY_PROVIDER` but server still fails to sync.

```
docker run --rm --name syncserver \
  -p 5000:5000 \
  -e SYNCSERVER_PUBLIC_URL=http://localhost:5000 \
  -e SYNCSERVER_IDENTITY_PROVIDER=http://localhost:3030 \
  -e SYNCSERVER_SECRET=5up3rS3kr1t \
  -e SYNCSERVER_SQLURI=sqlite:////tmp/syncserver.db \
  -e SYNCSERVER_BATCH_UPLOAD_ENABLED=true \
  -e SYNCSERVER_FORCE_WSGI_ENVIRON=true \
  mozilla/syncserver:latest \
  /usr/local/bin/gunicorn --bind 0.0.0.0:5000 \
  syncserver.wsgi_app
```

A workaround was to change [this](https://github.com/mozilla/PyBrowserID/blob/master/browserid/supportdoc.py#L217) to http and run outside of docker.

### PR DESCRIPTION
**Pull request by rfk**
Fixes #112.

### CODE REVIEW

File: _scripts/syncserver.sh [hunk 1]
```diff
@@ -3,6 +3,8 @@
 docker run --rm --name syncserver \
   -p 5000:5000 \
   -e SYNCSERVER_PUBLIC_URL=http://localhost:5000 \
+  -e SYNCSERVER_IDENTITY_PROVIDER=http://host.docker.internal:3030 \
+  -e SYNCSERVER_BROWSERID_VERIFIER=http://host.docker.internal:5050 \
```
Comment (rfk) [inline _scripts/syncserver.sh]: This has to make outgoing calls to the locally-running verifier and auth-server, but `--net=host` doesn't work in docker for windows.  The above works for me, but will need testing on other folks machines as well.

Comment (shane-tomlinson) [reply _scripts/syncserver.sh]: I was trying to get this to work locally and had different values that obviously didn't work!

**Comment by shane-tomlinson** [comment]
@rfk - I see this when I apply this patch and start the sync server:

> | ConnectionError: HTTPConnectionPool(host='host.docker.internal', port=3030): Max retries exceeded with url: /.well-known/fxa-client-configuration (Caused by NewConnectionError('<requests.packages.urllib3.connection.HTTPConnection object at 0x7f7ca3fc8750>: Failed to establish a new connection: [Errno 111] Connection refused',))
13|sync se | [2018-07-17 14:45:33 +0000] [12] [INFO] Worker exiting (pid: 12)
13|sync se | [2018-07-17 14:45:33 +0000] [7] [INFO] Shutting down: Master
13|sync se | [2018-07-17 14:45:33 +0000] [7] [INFO] Reason: Worker failed to boot.
13|sync se | [2018-07-17 14:46:01 +0000] [7] [INFO] Starting gunicorn 19.6.0
13|sync se | [2018-07-17 14:46:01 +0000] [7] [INFO] Listening at: http://0.0.0.0:5000 (7)
13|sync se | [2018-07-17 14:46:01 +0000] [7] [INFO] Using worker: sync
13|sync se | [2018-07-17 14:46:01 +0000] [12] [INFO] Booting worker with pid: 12
13|sync se | [2018-07-17 14:46:02 +0000] [12] [ERROR] Exception in worker process
13|sync se | Traceback (most recent call last):
13|sync se |   File "/usr/local/lib/python2.7/site-packages/gunicorn/arbiter.py", line 557, in spawn_worker
13|sync se |     worker.init_process()
13|sync se |   File "/usr/local/lib/python2.7/site-packages/gunicorn/workers/base.py", line 126, in init_process
13|sync se |     self.load_wsgi()
13|sync se |   File "/usr/local/lib/python2.7/site-packages/gunicorn/workers/base.py", line 136, in load_wsgi
13|sync se |     self.wsgi = self.app.wsgi()
13|sync se |   File "/usr/local/lib/python2.7/site-packages/gunicorn/app/base.py", line 67, in wsgi
13|sync se |     self.callable = self.load()
13|sync se |   File "/usr/local/lib/python2.7/site-packages/gunicorn/app/wsgiapp.py", line 65, in load
13|sync se |     return self.load_wsgiapp()
13|sync se |   File "/usr/local/lib/python2.7/site-packages/gunicorn/app/wsgiapp.py", line 52, in load_wsgiapp
13|sync se |     return util.import_app(self.app_uri)
13|sync se |   File "/usr/local/lib/python2.7/site-packages/gunicorn/util.py", line 357, in import_app
13|sync se |     __import__(module)
13|sync se |   File "/app/syncserver/wsgi_app.py", line 2, in <module>
13|sync se |     application = syncserver.main()
13|sync se |   File "/app/syncserver/__init__.py", line 245, in main
13|sync se |     config = get_configurator(global_config, **settings)
13|sync se |   File "/app/syncserver/__init__.py", line 237, in get_configurator
13|sync se |     config.include(includeme)
13|sync se |   File "/usr/local/lib/python2.7/site-packages/pyramid/config/__init__.py", line 754, in include
13|sync se |     c(configurator)
13|sync se |   File "/app/syncserver/__init__.py", line 137, in includeme
13|sync se |     config.include("tokenserver", route_prefix="/token")
13|sync se |   File "/usr/local/lib/python2.7/site-packages/pyramid/config/__init__.py", line 754, in include
13|sync se |     c(configurator)
13|sync se |   File "/usr/local/lib/python2.7/site-packages/tokenserver/__init__.py", line 46, in includeme
13|sync se |     load_and_register("oauth", config)
13|sync se |   File "/usr/local/lib/python2.7/site-packages/mozsvc/plugin.py", line 85, in load_and_register
13|sync se |     plugin = load_from_settings(section_name, settings)
13|sync se |   File "/usr/local/lib/python2.7/site-packages/mozsvc/plugin.py", line 132, in load_from_settings
13|sync se |     return klass(**kwargs)
13|sync se |   File "/usr/local/lib/python2.7/site-packages/tokenserver/verifiers.py", line 201, in __init__
13|sync se |     r = requests.get(server_url[:-3] + '/config')
13|sync se |   File "/usr/local/lib/python2.7/site-packages/requests/api.py", line 70, in get
13|sync se |     return request('get', url, params=params, **kwargs)
13|sync se |   File "/usr/local/lib/python2.7/site-packages/requests/api.py", line 56, in request
13|sync se |     return session.request(method=method, url=url, **kwargs)
13|sync se |   File "/usr/local/lib/python2.7/site-packages/requests/sessions.py", line 488, in request
13|sync se |     resp = self.send(prep, **send_kwargs)
13|sync se |   File "/usr/local/lib/python2.7/site-packages/requests/sessions.py", line 609, in send
13|sync se |     r = adapter.send(request, **kwargs)
13|sync se |   File "/usr/local/lib/python2.7/site-packages/requests/adapters.py", line 487, in send
13|sync se |     raise ConnectionError(e, request=request)
13|sync se | ConnectionError: HTTPConnectionPool(host='127.0.0.1', port=9010): Max retries exceeded with url: /config (Caused by NewConnectionError('<requests.packages.urllib3.connection.HTTPConnection object at 0x7fd6c73f9590>: Failed to establish a new connection: [Errno 111] Connection refused',))
13|sync se | [2018-07-17 14:46:02 +0000] [12] [INFO] Worker exiting (pid: 12)
13|sync se | [2018-07-17 14:46:02 +0000] [7] [INFO] Shutting down: Master
13|sync se | [2018-07-17 14:46:02 +0000] [7] [INFO] Reason: Worker failed to boot.

**Comment by shane-tomlinson** [comment]
> @rfk - I see this when I apply this patch and start the sync server:

I expect this is because I need to download the docker image for https://github.com/mozilla-services/syncserver/pull/127

File: _scripts/install_all.sh [hunk 2]
```diff
@@ -16,7 +16,7 @@ git clone https://github.com/mozilla/fxa-auth-db-mysql.git &
 
 git clone https://github.com/mozilla/fxa-customs-server.git &
 
-git clone https://github.com/vladikoff/browserid-verifier.git -b http &
+git clone https://github.com/mozilla/browserid-verifier.git &
```
Comment (vladikoff) [inline _scripts/install_all.sh]: 🥇

**Comment by vladikoff** [comment]
Checking logs on travis, it says: 

image

**Comment by rfk** [comment]
> Checking logs on travis, it says:

:-( I wonder if the version of docker on travis does not support the special `host.docker.internal` name, IIRC it's been in docker-for-windows and docker-for-mac for a while but only recently added to docker-for-linux.

**Comment by rfk** [comment]
> $ docker --version
> Docker version 18.05.0-ce, build f150324

**Comment by rfk** [comment]
Oh, hrm, a comment went missing here - it turns out I was wrong, it doesn't look like "host.docker.internal" has landed in docker-for-linux yet.

**Comment by rfk** [comment]
I fixed travis with a bit of if-then-else to choose the right localhost addr.  I removed the setting of `SYNCSERVER_IDENTITY_PROVIDER` which was causing it to block waiting for content-server to start, because I don't think this should be necessary for development.  (The syncserver will accept assertions from *any* browserid IdP in this case, which is fine for dev purposes).

File: .travis.yml [hunk 3]
```diff
@@ -18,6 +18,7 @@ addons:
     - ubuntu-toolchain-r-test
     packages:
     - g++-4.8
+    - docker-ce
```
Comment (rfk) [inline .travis.yml]: This may not be necessary anymore, but seemed harmless to leave in just for explicitness.

**Comment by vladikoff** [comment]
We are Syncing yet again!

**Comment by shane-tomlinson** [comment]
:( Am I doing something wrong? I'm not.

> 13|sync se | WARNING: Published ports are discarded when using host network mode
13|sync se | [2018-07-18 14:12:10 +0000] [7] [INFO] Starting gunicorn 19.6.0
13|sync se | [2018-07-18 14:12:10 +0000] [7] [INFO] Listening at: http://0.0.0.0:5000 (7)
13|sync se | [2018-07-18 14:12:10 +0000] [7] [INFO] Using worker: sync
13|sync se | [2018-07-18 14:12:10 +0000] [12] [INFO] Booting worker with pid: 12
13|sync se | [2018-07-18 14:12:11 +0000] [12] [ERROR] Exception in worker process
13|sync se | Traceback (most recent call last):
13|sync se |   File "/usr/local/lib/python2.7/site-packages/gunicorn/arbiter.py", line 557, in spawn_worker
13|sync se |     worker.init_process()
13|sync se |   File "/usr/local/lib/python2.7/site-packages/gunicorn/workers/base.py", line 126, in init_process
13|sync se |     self.load_wsgi()
13|sync se |   File "/usr/local/lib/python2.7/site-packages/gunicorn/workers/base.py", line 136, in load_wsgi
13|sync se |     self.wsgi = self.app.wsgi()
13|sync se |   File "/usr/local/lib/python2.7/site-packages/gunicorn/app/base.py", line 67, in wsgi
13|sync se |     self.callable = self.load()
13|sync se |   File "/usr/local/lib/python2.7/site-packages/gunicorn/app/wsgiapp.py", line 65, in load
13|sync se |     return self.load_wsgiapp()
13|sync se |   File "/usr/local/lib/python2.7/site-packages/gunicorn/app/wsgiapp.py", line 52, in load_wsgiapp
13|sync se |     return util.import_app(self.app_uri)
13|sync se |   File "/usr/local/lib/python2.7/site-packages/gunicorn/util.py", line 357, in import_app
13|sync se |     __import__(module)
13|sync se |   File "/app/syncserver/wsgi_app.py", line 2, in <module>
13|sync se |     application = syncserver.main()
13|sync se |   File "/app/syncserver/__init__.py", line 257, in main
13|sync se |     config = get_configurator(global_config, **settings)
13|sync se |   File "/app/syncserver/__init__.py", line 249, in get_configurator
13|sync se |     config.include(includeme)
13|sync se |   File "/usr/local/lib/python2.7/site-packages/pyramid/config/__init__.py", line 754, in include
13|sync se |     c(configurator)
13|sync se |   File "/app/syncserver/__init__.py", line 62, in includeme
13|sync se |     r = requests.get(urljoin(idp, '/.well-known/fxa-client-configuration'))
13|sync se |   File "/usr/local/lib/python2.7/site-packages/requests/api.py", line 70, in get
13|sync se |     return request('get', url, params=params, **kwargs)
13|sync se |   File "/usr/local/lib/python2.7/site-packages/requests/api.py", line 56, in request
13|sync se |     return session.request(method=method, url=url, **kwargs)
13|sync se |   File "/usr/local/lib/python2.7/site-packages/requests/sessions.py", line 488, in request
13|sync se |     resp = self.send(prep, **send_kwargs)
13|sync se |   File "/usr/local/lib/python2.7/site-packages/requests/sessions.py", line 609, in send
13|sync se |     r = adapter.send(request, **kwargs)
13|sync se |   File "/usr/local/lib/python2.7/site-packages/requests/adapters.py", line 487, in send
13|sync se |     raise ConnectionError(e, request=request)
13|sync se | ConnectionError: HTTPConnectionPool(host='localhost', port=3030): Max retries exceeded with url: /.well-known/fxa-client-configuration (Caused by NewConnectionError('<requests.packages.urllib3.connection.HTTPConnection object at 0x7f2e2713a990>: Failed to establish a new connection: [Errno 111] Connection refused',))
13|sync se | [2018-07-18 14:12:11 +0000] [12] [INFO] Worker exiting (pid: 12)
13|sync se | [2018-07-18 14:12:11 +0000] [7] [INFO] Shutting down: Master
13|sync se | [2018-07-18 14:12:11 +0000] [7] [INFO] Reason: Worker failed to boot.

I have git pulled, & run update_all

**Comment by rfk** [comment]
Hrm...

>  ConnectionError: HTTPConnectionPool(host='localhost', port=3030):
> Max retries exceeded with url: /.well-known/fxa-client-configuration

So this is trying to connect to `http://localhost:3030/.well-known/fxa-client-configuration`, but I don't know why.  The latst version of `./_scripts/syncserver.sh` should not be setting the env var for that.  Can you double-check any references to localhost in that script.  I would expect only the following:

```
$> grep 'localhost' syncserver.sh
  -e SYNCSERVER_PUBLIC_URL=http://localhost:5000 \
```

**Comment by shane-tomlinson** [comment]
> grep localhost syncserver.sh
  
-e SYNCSERVER_PUBLIC_URL=http://localhost:5000 \

**Pull request merged**